mlir/lib/Util/Dependency.cpp

```diff
@@ -963,12 +963,17 @@ getAllReadAccessedMemrefOperandsFromOp(Operation *op) {
         std::get<2>(entry.second) = strides;
         return entry;
       };
-  auto pushMemrefEntryToVector = []<typename T>(T entry,
-                                                SmallVector<T> &vector) {
-    if (!isa<MemRefType>(entry.first.getType()))
-      return;
-    vector.push_back(entry);
-  };
```

Comment:
On the other hand you could just use some C++14 Python-esque code like
```c++
 auto pushMemrefEntryToVector = [](auto entry, auto &vector) {
    if (!isa<MemRefType>(entry.first.getType()))
      return;
    vector.push_back(entry);
  };
```

Reply:
That works! Thanks for the comment, @keryell!